### PR TITLE
fix: Make sure form fields' dropdown menus are positioned above buttons in other form fields when the window loses focus

### PR DESCRIPTION
**Pull request by lerouxb**
It fixes this problem:

eek

Open that selectbox's dropdown then open devtools so the main window area loses focus and then buttons from other fields end up sitting on top of the open dropdown.

### CODE REVIEW

File: packages/compass-components/src/components/form-field-container.tsx [hunk 1]
```diff
@@ -4,6 +4,10 @@ import React from 'react';
 
 const formFieldContainerStyles = css({
   margin: `${spacing[4]}px 0`,
+  // This is to prevent buttons from other form fields being positioned on top
+  // of a select dropdown menu in this form field when the window loses focus.
```
Comment (Anemy) [inline packages/compass-components/src/components/form-field-container.tsx:8]: Is/was this a LeafyGreen issue? Might be worth seeing if the new versions (if we aren't on newest) behave the same. Looks like an e2e test is failing. Changes are fine by me. lgtm once green

Comment (lerouxb) [reply packages/compass-components/src/components/form-field-container.tsx:8]: I don't know.

File: packages/compass-e2e-tests/helpers/commands/add-collection.ts [hunk 2]
```diff
@@ -70,15 +95,24 @@ export async function addCollection(
...
       await span.waitForDisplayed();
+      await waitForAnimations(browser, span as Element<'async'>);
+      await browser.screenshot(`custom-collation-${key}.png`);
       await span.click();
+      await menu.waitForDisplayed({ reverse: true });
+
+      const button = await browser.$(
+        Selectors.createCollectionCustomCollationFieldButton(key)
+      );
+      console.log({ [key]: await button.getText() });
```
Comment (Anemy) [inline packages/compass-e2e-tests/helpers/commands/add-collection.ts:115]:
```suggestion
```

File: packages/compass-e2e-tests/helpers/commands/add-collection.ts [hunk 3]
```diff
@@ -31,6 +33,29 @@ export type AddCollectionOptions = {
...
+
+    const result = {
+      ...(await element.getLocation()),
+      ...(await element.getSize()),
+    };
+    const stopped = _.isEqual(result, previousResult);
+    previousResult = result;
+    return stopped;
+  });
+}
```
Comment (Anemy) [inline packages/compass-e2e-tests/helpers/commands/add-collection.ts:57]: Do we want to move this into the file https://github.com/mongodb-js/compass/blob/93307133a16a7320bd205cd04424424bbbadaaa1/packages/compass-e2e-tests/helpers/commands/wait-for-animations.ts
Maybe we can reduce the duplication there too.

**Pull request closed without merging**